board/safety/safety_gm.h

```diff
@@ -218,12 +236,44 @@ static int gm_tx_hook(CANPacket_t *to_send, bool longitudinal_allowed) {
     }
   }
 
+  // BUTTONS: used for resume spamming and cruise cancellation with stock longitudinal
+  if ((addr == 481) && (gm_hw == GM_CAM)) {
+    int button = (GET_BYTE(to_send, 5) >> 4) & 0x7U;
+
+    bool allowed_cancel = (button == 6) && cruise_engaged_prev;
+    if (!allowed_cancel) {
+      tx = 0;
+    }
+  }
+
   // 1 allows the message through
   return tx;
 }
 
+static int gm_fwd_hook(int bus_num, CANPacket_t *to_fwd) {
+
+  int bus_fwd = -1;
+
+  if (gm_hw != GM_ASCM) {
```

Comment:
any reason this is written like this? reads better as:
```suggestion
  if (gm_hw != GM_CAM) {
```

Reply:
I think it was because there was going to be a GM_CAM_LONG, but we can figure that out later in another PR. That should probably be another variable like the other makes